### DISCUSSION TITLE
Upload files

### DISCUSSION THREAD

**Discussion by artola** [enisdenjo/graphql-ws#297] [Q&A] [answered]
I am using graphql-ws, awesome!
I did it starting for subscriptions with Relay, later on I added queries and mutations too, but at this point I am trying to upload files (multipart request), is it planned to add support for them in the library?

**Comment by enisdenjo** [comment, marked as answer]
Multipart requests are HTTP requests, nothing to do with WebSockets. Regardless, I don't recommend doing file uploads over WebSockets (and I never tried either). Since HTTP is a far better medium here, `graphql-ws` will probably never support file uploads. Consider using HTTP, as you hinted, instead.
